sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/view/AdviceAttributesProcessor.java

```diff
@@ -23,11 +23,28 @@ final class AdviceAttributesProcessor extends AttributesProcessor {
 
   @Override
   public Attributes process(Attributes incoming, Context context) {
+    // Exit early to avoid allocations if the incoming attributes do not have extra keys to be
+    // filtered
+    if (!hasExtraKeys(incoming)) {
+      return incoming;
+    }
     AttributesBuilder builder = incoming.toBuilder();
     builder.removeIf(key -> !attributeKeys.contains(key));
     return builder.build();
   }
 
+  /** Returns true if {@code attributes} has keys not contained in {@link #attributeKeys}. */
+  private boolean hasExtraKeys(Attributes attributes) {
+    boolean[] result = {false};
```

Comment:
This little trick allows you to update state from inside a lambda without an atomic variable.